### ISSUE TITLE
Assumed PSP format

### ISSUE THREAD

**Issue by CarrieFilion** [michael-petersen/exptool#22]
There are a few places in the code that assume that the file format is PSP, and have no flexibility for SPL and/or legacy format. I think the easiest fix here is just replacing the call to io.psp_io.Input() with a call to io.particle.Input(). Below, I've listed the files that explicitly call PSP_io. 

Explicit psp_io calls:
- basis/potential.py
- orbits/orbit.py
- utils/integrate.py

There also appears to be different formats for the outcoef files, the SPL outcoef files cannot be read with the functionalities within basis/eof.py.  SPL outcoef files can be read with io.outcoef. read_binary_eof_coefficients_old, but there doesn't appear to be an equivalent function for eof.read_binary_eof_coefficients_dict.

**Comment by michael-petersen** [comment]
I adjusted the `psp_io` calls to be `particle` calls, and added a new parameter `fileprefix` that can handle `SPL` or `OUT` (user must specify). I tested, but not thoroughly (we need a test suite!).

**Comment by michael-petersen** [comment]
There is now a reader that should be able to reproduce the functionality of `eof.read_binary_eof_coefficients_dict`:
```C = outcoef.OutCoef()
# this will call (you don't need to use this):
C.read_binary_eof_coefficients_old()
# and then you may call
C._repackage_cylindrical_coefficients_compatibility()
```
If you run the last line, it will produce the same output as `eof.read_binary_eof_coefficients_dict`.

However, it might be best for us/me to adjust whatever the upstream problem is -- probably bar position angle determination?

**Comment by CarrieFilion** [comment]
The issue with the `eof.read_binary_eof_coefficients_dict` came up in trying to do the workflow for a bar file (`read_binary_eof_coefficients_dict` -> `calculate_eof_phase` -> `print_eof_barfile`), and in trying to do the `quick_plot_coefs`. 

I pulled the recent changes, but running
`C = outcoef.OutCoef('sim_directory/outcoef.star.run2Fa')
C.repackage_cylindrical_coefficients_compatibility()` 
comes up with an error on line 478 in io.outcoef ("IndexError: index 7 is out of bounds for axis 2 with size 7"). I tried replacing line 476 with `for mm in range(0,numm):`, which allowed the repackaging to run, but then I hit an error when I use the output of `C._repackage_cylindrical_coefficients_compatibility()`as the input to run `eof.calculate_eof_phase()` (line line 1990, in calculate_eof_phase "mmax=EOFDict[0].mmax KeyError: 0").

**Comment by michael-petersen** [comment]
Thanks for the info, very clear!

For the m range issue, I had checked in a bug confusing myself with mmax (the maximum Fourier harmonic), with the _number_ of Fourier harmonics, which is mmax+1. I fixed this.

For the phase calculation, I had not fixed a bug that assumed the first key would be named '0'. The key is really a float with the timestamp, which might not be a long term solution anyway.

**Comment by CarrieFilion** [comment]
The repackaging works now, but unfortunately, `calculate_eof_phase` does not now. I've pasted the error message below. The warning is maybe related to the weird angles that I mention in Issue #21 . This error comes up in the workflow `C = outcoef.OutCoef('sim_directory/outcoef.star.ru2Fa` -> `EOF2Dict = C.repackage_cylindrical_coefficients_compatibility()` -> `DCp = eof.calculate_eof_phase(EOF2Dict)`. 

```
unwrap_phases WARNING: Were the phases calculated with arctan2?
Traceback (most recent call last):
  File "exploring.py", line 41, in <module>
    DCp = eof.calculate_eof_phase(EOF2Dict)
  File "/home/filion/martinsims/exptools/exptool/exptool/basis/eof.py", line 2077, in calculate_eof_phase
    DC['speed'][mm][:,nn] = np.ediff1d(utils.savitzky_golay(DC['unphase'][mm][:,nn],smooth_box,smooth_order),to_begin=0.)/np.ediff1d(DC['time'],to_begin=100.)
ValueError: could not broadcast input array from shape (101) into shape (1)
```


If I instead set filter to false (`DCp = eof.calculate_eof_phase(EOF2Dict,filter=False)`),  I get:

```
unwrap_phases WARNING: Were the phases calculated with arctan2?
Traceback (most recent call last):
  File "exploring.py", line 41, in <module>
    DCp = eof.calculate_eof_phase(EOF2Dict,filter=False)
  File "/home/filion/martinsims/exptools/exptool/exptool/basis/eof.py", line 2083, in calculate_eof_phase
    DC['speed'][mm][0,nn] = DC['speed'][mm][1,nn]
IndexError: index 1 is out of bounds for axis 0 with size 1
```

**Comment by michael-petersen** [comment]
I had inadvertently introduced _another_ bug to the `_compatibility` version of the coefficients, that was stopping any keys (simulation times) from being properly logged. So that explains the behaviour above. I tested on an old set of coefficients; I need to generate a new set to try the workflow again.

**Comment by CarrieFilion** [comment]
Ok - as of your most recent push, I could generate bar files for both PSP and SPL format. I hit a small snag with file formats again when trying to run `potential.get_fields()`. I made two small tweaks that I think I fixed the issue, and have pushed the changes to carrie_edits. I'm putting together example notebooks for a few workflows this week, and will be sure to check that the notebooks work with both PSP and SPL format. I'll let ya know if I run into issues, and I'll push the notebooks to carrie_edits when I'm done!
